internal/controller/storageclient_controller.go

```diff
@@ -697,15 +721,48 @@ func removeStorageClaimAsOwner(obj client.Object) {
 	}
 }
 
-func (r *storageClientReconcile) reconcileResource(obj client.Object, rawObject []byte) error {
-	objectMeta := &metav1.PartialObjectMetadata{}
-	if err := json.Unmarshal(rawObject, objectMeta); err != nil {
-		return err
+func (r *storageClientReconcile) reconcileResourcesByGVK(
+	obj client.Object,
+	reqs []reconcileRequest,
```

Comment:
These are not requests, these are the desired states for some of the object that we need to reconcile. 
Please remember that a result of delete is still considered a reconcile operation

Reply:
changed to `desiredStates`